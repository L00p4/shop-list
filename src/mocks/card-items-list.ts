export const mockCardItemsList = [
  {
    title: 'Lista de Compras',
    itemsCount: 12,
    creationDate: '01/01/2023',
    lastPurchaseDate: undefined,
    onEditClick: () => alert('Editar lista'),
    onBuyClick: () => alert('Comprar itens')
  },
  {
    title: 'Churrasco',
    itemsCount: 5,
    creationDate: '02/02/2023',
    lastPurchaseDate: '05/05/2023',
    onEditClick: () => alert('Editar lista'),
    onBuyClick: () => alert('Comprar itens')
  }
]
