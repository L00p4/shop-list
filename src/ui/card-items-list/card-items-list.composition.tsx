import CardItemsList, { CardItemsListSkeleton } from '.'

export const Primary = () => <CardItemsList />

export const Skeleton = () => <CardItemsListSkeleton />
