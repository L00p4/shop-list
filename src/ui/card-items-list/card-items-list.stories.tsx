import { Meta, StoryObj } from '@storybook/react'

import CardItemsList, { CardItemsListSkeleton } from '.'

const meta: Meta<typeof CardItemsList> = {
  title: 'CardItemsList',
  component: CardItemsList,
  tags: ['autodocs'],
  args: {},
  parameters: {
    layout: 'centered',
    backgrounds: {
      default: 'dark'
    }
  }
}

type Story = StoryObj<typeof CardItemsList>

export const Primary: Story = {
  args: {
    title: 'Lista de Compras',
    itemsCount: 12,
    lastPurchaseDate: '10/10/2023',
    creationDate: '01/01/2023',
    onEditClick: () => alert('Editar lista'),
    onBuyClick: () => alert('Comprar itens')
  }
}

export const Skeleton: Story = {
  render: () => <CardItemsListSkeleton />,
  args: {}
}

export default meta
