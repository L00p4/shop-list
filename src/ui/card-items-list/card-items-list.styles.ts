import styled from 'styled-components'

export const WrapperCardItemsList = styled.div`
  display: flex;
  flex-direction: column;
  gap: var(--space-4);

  padding: var(--space-3);
  border-radius: var(--radius-md);
  border-left: var(--space-1) solid var(--purple-500);

  background-color: var(--bg-tertiary);

  h2 {
    color: var(--text-primary);
    font-size: var(--font-size-xl);
    font-weight: var(--font-weight-medium);
  }
`

export const ContentCardItemContainer = styled.div`
  display: flex;
  justify-content: space-between;
  gap: var(--space-4);
  align-items: center;
`

export const ContentCardItem = styled.div`
  display: flex;
  flex-direction: column;

  color: var(--text-tertiary);
  font-size: var(--font-size-xs);
  font-weight: var(--font-weight-extralight);

  p.items-count {
    font-size: var(--font-size-sm);
  }
`
export const ButtonContainerCardItem = styled.div`
  display: flex;
  flex-direction: column;
  gap: var(--space-2);
`
