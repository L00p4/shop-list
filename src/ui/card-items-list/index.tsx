import {
  ButtonContainerCardItem,
  ContentCardItem,
  WrapperCardItemsList,
  ContentCardItemContainer
} from './card-items-list.styles'
import Button from '../button'

export { default as CardItemsListSkeleton } from './card-items-list.skeleton'

type CardItemsListProps = {
  title: string
  itemsCount: number
  lastPurchaseDate: string
  creationDate: string
  onEditClick: () => void
  onBuyClick: () => void
}

const CardItemsList = ({
  title,
  itemsCount,
  lastPurchaseDate,
  creationDate,
  onEditClick,
  onBuyClick
}: CardItemsListProps) => (
  <WrapperCardItemsList>
    <h2>{title}</h2>
    <ContentCardItemContainer>
      <ContentCardItem>
        <p className="items-count">{itemsCount} itens</p>
        <p>Ãšltima compra {lastPurchaseDate}</p>
        <p>Criada em {creationDate}</p>
      </ContentCardItem>
      <ButtonContainerCardItem>
        <Button onClick={onEditClick} variant="primary" size="compact">
          ver/edit
        </Button>
        <Button onClick={onBuyClick} variant="secondary" size="compact">
          comprar
        </Button>
      </ButtonContainerCardItem>
    </ContentCardItemContainer>
  </WrapperCardItemsList>
)

export default CardItemsList
