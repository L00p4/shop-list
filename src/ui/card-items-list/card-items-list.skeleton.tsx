import { WrapperCardItemsList } from './card-items-list.styles'

const CardItemsListSkeleton = () => (
  <WrapperCardItemsList>
    <h1 className="skeleton">CardItemsList</h1>
  </WrapperCardItemsList>
)

export default CardItemsListSkeleton
