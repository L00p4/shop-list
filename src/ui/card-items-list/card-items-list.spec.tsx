import '@testing-library/jest-dom'
import 'jest-environment-jsdom'
import 'jest-styled-components'

import { render, screen } from '@testing-library/react'

import CardItemsList from '.'
import { Primary, Skeleton } from './card-items-list.composition'

describe('<CardItemsList />', () => {
  it('should render the heading', () => {
    render(<CardItemsList />)

    expect(
      screen.getByRole('heading', { name: /nome da lista aqui/i })
    ).toBeInTheDocument()
  })

  it('should render Primary', () => {
    expect(render(<Primary />).container).toMatchSnapshot()
  })

  it('should render Skeleton', () => {
    expect(render(<Skeleton />).container).toMatchSnapshot()
  })
})
