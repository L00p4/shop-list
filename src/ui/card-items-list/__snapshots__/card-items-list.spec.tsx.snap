// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<CardItemsList /> should render Primary 1`] = `
.c0 {
  display: flex;
  flex-direction: column;
  gap: var(--space-4);
  padding: var(--space-3);
  border-radius: var(--radius-md);
  border-left: var(--space-1) solid var(--purple-500);
  background-color: var(--bg-tertiary);
}

.c0 h2 {
  color: var(--text-primary);
  font-size: var(--font-size-xl);
  font-weight: var(--font-weight-medium);
}

.c1 {
  display: flex;
  justify-content: space-between;
  gap: var(--space-4);
  align-items: center;
}

.c2 {
  display: flex;
  flex-direction: column;
  color: var(--text-tertiary);
  font-size: var(--font-size-xs);
  font-weight: var(--font-weight-extralight);
}

.c2 p.items-count {
  font-size: var(--font-size-sm);
}

.c3 {
  display: flex;
  flex-direction: column;
  gap: var(--space-2);
}

.c4 {
  border: none;
  outline: none;
  cursor: pointer;
  font-family: inherit;
  font-weight: var(--font-medium);
  display: inline-flex;
  align-items: center;
  justify-content: center;
  gap: var(--space-2);
  transition: all var(--transition-normal);
  background: var(--purple-500);
  color: var(--text-primary);
  border: 1px solid var(--purple-500);
  padding: var(--space-1);
  font-size: var(--font-size-xs);
  border-radius: var(--radius-sm);
  user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  touch-action: manipulation;
  -webkit-tap-highlight-color: transparent;
}

.c4:hover:not(:disabled) {
  background: var(--purple-700);
  border-color: var(--purple-700);
}

.c4:active:not(:disabled) {
  background: var(--purple-700);
  transform: translateY(1px);
}

.c4:disabled {
  opacity: 0.6;
  cursor: not-allowed;
  transform: none;
}

.c4:focus-visible {
  outline: 2px solid var(--purple-500);
  outline-offset: 2px;
}

.c5 {
  border: none;
  outline: none;
  cursor: pointer;
  font-family: inherit;
  font-weight: var(--font-medium);
  display: inline-flex;
  align-items: center;
  justify-content: center;
  gap: var(--space-2);
  transition: all var(--transition-normal);
  background: var(--border-default);
  color: var(--text-secondary);
  border: 1px solid var(--border-default);
  padding: var(--space-1);
  font-size: var(--font-size-xs);
  border-radius: var(--radius-sm);
  user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  touch-action: manipulation;
  -webkit-tap-highlight-color: transparent;
}

.c5:hover:not(:disabled) {
  background: var(--border-default);
  border-color: var(--text-tertiary);
}

.c5:active:not(:disabled) {
  background: var(--border-default);
  transform: translateY(1px);
}

.c5:disabled {
  opacity: 0.6;
  cursor: not-allowed;
  transform: none;
}

.c5:focus-visible {
  outline: 2px solid var(--purple-500);
  outline-offset: 2px;
}

<div>
  <div
    class="c0"
  >
    <h2 />
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <p
          class="items-count"
        >
           itens
        </p>
        <p>
          Ãšltima compra 
        </p>
        <p>
          Criada em 
        </p>
      </div>
      <div
        class="c3"
      >
        <button
          class="c4"
          type="button"
          variant="primary"
        >
          ver/edit
        </button>
        <button
          class="c5"
          type="button"
          variant="secondary"
        >
          comprar
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<CardItemsList /> should render Skeleton 1`] = `
.c0 {
  display: flex;
  flex-direction: column;
  gap: var(--space-4);
  padding: var(--space-3);
  border-radius: var(--radius-md);
  border-left: var(--space-1) solid var(--purple-500);
  background-color: var(--bg-tertiary);
}

.c0 h2 {
  color: var(--text-primary);
  font-size: var(--font-size-xl);
  font-weight: var(--font-weight-medium);
}

<div>
  <div
    class="c0"
  >
    <h1
      class="skeleton"
    >
      CardItemsList
    </h1>
  </div>
</div>
`;
